fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/MetricLeftUnionAvailability.java

```diff
@@ -53,13 +56,43 @@ public SimplifiedIntervalList getAvailableIntervals(PhysicalDataSourceConstraint
             return new SimplifiedIntervalList();
         }
 
+        Collection<String> representativeColumns = representativeAvailability.getAllAvailableIntervals().keySet();
+
+        Set<String> filteredMetricNames = dataSourceMetricNames.stream()
+                .filter(name -> representativeColumns.contains(name))
+                .collect(Collectors.toSet());
```

Comment:
Why do we need to take intersection between `dataSourceMetricNames` and `representativeColumns`?